crates/router/src/core/payment_methods/cards.rs

```diff
@@ -895,14 +1018,32 @@ pub async fn get_client_secret_or_add_payment_method(
 }
 
 #[cfg(all(feature = "v2", feature = "payment_methods_v2"))]
-#[instrument(skip_all)]
-pub async fn get_client_secret_or_add_payment_method(
-    _state: &routes::SessionState,
-    _req: api::PaymentMethodCreate,
-    _merchant_account: &domain::MerchantAccount,
-    _key_store: &domain::MerchantKeyStore,
-) -> errors::RouterResponse<api::PaymentMethodResponse> {
-    todo!()
+pub async fn create_pm_additional_data_update(
+    pmd: &api::PaymentMethodCreateData,
+    state: &routes::SessionState,
+    key_store: &domain::MerchantKeyStore,
+    vault_id: Option<String>,
+    payment_method: Option<api_enums::PaymentMethod>,
+    payment_method_type: Option<api_enums::PaymentMethodType>,
+) -> errors::RouterResult<storage::PaymentMethodUpdate> {
+    let card = match pmd.clone() {
+        api::PaymentMethodCreateData::Card(card) => PaymentMethodsData::Card(card.into()),
+    };
+
+    let pmd: Encryptable<Secret<serde_json::Value>> = create_encrypted_data(state, key_store, card)
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)
+        .attach_printable("Unable to encrypt Payment method data")?;
+
+    let pm_update = storage::PaymentMethodUpdate::AdditionalDataUpdate {
```

Comment:
```suggestion
    let pm_update = storage::PaymentMethodUpdate::VaultIdUpdate{
```

Reply:
The variant actually takes in a bunch of stuff to update and not just the Vault Id right?